### ISSUE TITLE
`jax.nn.softmax` default JVP violates logit-translation invariance in a finite float64 tail

### ISSUE THREAD

**Issue by beanduan22** [jax-ml/jax#40344]
### Description

The derivative of `jax.nn.softmax` does not preserve its forward translation invariance at a finite `float64` input.

```text
softmax(z + c * 1) = softmax(z)
```

Thus the VJP with respect to logits must sum to zero. For `z = [40.0, 0.0]`, the VJP of `softmax(z)[0]` sums to `-4.248354255291589e-18` instead. The missing first component is finite and representable: the expected VJP is approximately `[+4.248354255291589e-18, -4.248354255291589e-18]`.

```python
import math
import jax
import jax.numpy as jnp

jax.config.update("jax_enable_x64", True)
logits = jnp.asarray([40.0, 0.0], dtype=jnp.float64)
softmax_first = lambda x: jax.nn.softmax(x)[0]
cross_entropy = lambda x: -jax.nn.log_softmax(x)[0]
softmax_grad = jax.grad(softmax_first)(logits)
cross_entropy_grad = jax.grad(cross_entropy)(logits)
print("softmax VJP:", softmax_grad.tolist())
print("softmax VJP sum:", float(softmax_grad.sum()))
print("cross entropy VJP:", cross_entropy_grad.tolist())
print("cross entropy VJP sum:", float(cross_entropy_grad.sum()))
print("expected tail magnitude:", math.exp(-40.0) / (1.0 + math.exp(-40.0)) ** 2)
```

## Actual result

On JAX `0.9.2`, CPU and two CUDA devices reproduce:

```text
softmax VJP: [0.0, -4.248354255291589e-18]
softmax VJP sum: -4.248354255291589e-18
cross entropy VJP: [0.0, 4.248354255291589e-18]
cross entropy VJP sum: 4.248354255291589e-18
expected tail magnitude: 4.248354255291589e-18
```

## Expected result

Both VJPs should sum to zero. The expected cross-entropy gradient is the negation of the expected softmax VJP.

### System info (python version, jaxlib version, accelerator, etc.)

```text
JAX version: 0.9.2
jaxlib version: 0.9.2
Python version: 3.13.5
Input dtype: float64, with jax_enable_x64=True
Devices tested: CPU and two CUDA GPUs
```

**Comment by jakevdp** [comment]
Thanks for the report – this looks to fall under https://github.com/jax-ml/jax/issues/38651.
